File: src/main/java/com/EmployeesREST/validation/EmployeeDateOfBirthValidator.java
package com.EmployeesREST.validation;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;
import java.time.LocalDate;

public class EmployeeDateOfBirthValidator implements
        ConstraintValidator<CustomDateValidator, LocalDate> {
    int daysBeforeBecomeSadAdult = 6575;
    @Override
    public void initialize(CustomDateValidator customDate) {
    }

    @Override
    public boolean isValid(LocalDate localDateToValidate, ConstraintValidatorContext cxt) {

        return localDateToValidate.plusDays(daysBeforeBecomeSadAdult).isBefore(LocalDate.now());
    }

}

File: src/main/korlin/com/employees/validation/EmployeeDateOfBirthValidator.kt
package com.employees.validation

import java.time.LocalDate
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class EmployeeDateOfBirthValidator : ConstraintValidator<CustomDateValidator?, LocalDate> {
    var daysBeforeBecomeSadAdult = 6575
    override fun initialize(customDate: CustomDateValidator?) {}
    override fun isValid(localDateToValidate: LocalDate, cxt: ConstraintValidatorContext): Boolean {
        return localDateToValidate.plusDays(daysBeforeBecomeSadAdult.toLong()).isBefore(LocalDate.now())
    }
}